### PR TITLE
Added Form Register

### PR DESCRIPTION
**Pull request by Tiaraha-funny**
Hey team,

I have created this form register branch to handle the registration component. I think t is ready to be reviewed so please have a look at it and leave any comment for it to be improved.

Click up link: https://app.clickup.com/t/41k87k
Refs: #41k87k 

Thanks a lot.

### CODE REVIEW

File: src/components/Buttons/Buttons.tsx [hunk 1]
```diff
@@ -2,77 +2,64 @@ import React from 'react'
 import styled from 'styled-components'
 import { fonts } from '../../globalStyles/fonts'
 
-export interface ButtonsProps {
+export interface ButtonProps {
   label?: string
   learnEnabled?: boolean
   disabled?: boolean
-  onClick?: () => void
+  onClick?: (event: React.FormEvent<HTMLFormElement>) => any
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:9]: Event will not always be a form event... For now its fine but will have to be changed in the future

File: src/components/Buttons/Buttons.tsx [hunk 2]
```diff
@@ -2,77 +2,64 @@ import React from 'react'
 import styled from 'styled-components'
 import { fonts } from '../../globalStyles/fonts'
 
-export interface ButtonsProps {
+export interface ButtonProps {
   label?: string
   learnEnabled?: boolean
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:7]: rename learnEnabled to isPrimary

File: src/components/InputName/InputName.tsx [hunk 3]
```diff
@@ -75,17 +82,63 @@ const InputElement = styled.input`
...
+  onChange,
+  errorMes,
+}) => {
   return (
     <InputContainer>
-      <label className='labelStyle'>{label}</label>
-      <InputElement placeholder={placeholder} className={name} />
+      <LabelStyle className='labelStyle'>
+        <div className='label'> {label}</div>
+        <div className='error'>{errorMes} </div>
```
Comment (andrzej-onja) [inline src/components/InputName/InputName.tsx:133]: No className assignments, pleases use styled components and pass props to them

File: src/mediaQuery/mediaQueries.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  ld: 900,
+  xl: 1000,
+}
+
+export const mediaQueries = (key: keyof typeof breakpoints) => {
+  return (style: TemplateStringsArray | String) => {
+    const maxWidth = key ? `(min-width: ${breakpoints[key]}px)` : ''
+    return `@media ${maxWidth} { ${style} }`
+  }
+}
```
Comment (andrzej-onja) [inline src/mediaQuery/mediaQueries.tsx:14]: we are using this version https://codesandbox.io/s/media-queries-in-styled-components-typescript-forked-o4tt0 . Please change it. Its based on the article but heavily modified

**Review by andrzej-onja — CHANGES_REQUESTED** [review]

**Review by lightme-fan — COMMENTED** [review]
Perhaps, you'd better fix Andrej's comments. Thanks

File: src/components/InputName/InputName.tsx [hunk 5]
```diff
@@ -14,10 +14,20 @@ const inputStyles = css`
...
-  text-transform: capitalize;
   outline: none;
   border: 1px solid #041d42;
   border-radius: 6px;
+
+  ::placeholder {
+    text-transform: capitalize;
+  }
+
+  .noPlaceholder {
```
Comment (lightme-fan) [inline src/components/InputName/InputName.tsx:25]: You shouldn't have a className

File: src/components/InputName/InputName.tsx [hunk 6]
```diff
@@ -33,6 +43,14 @@ const InputContainer = styled.div`
...
   }
+
+  .invalidEmail {
+    border-color: #fc462b;
+    ::placeholder {
+      color: #fc462b;
+    }
+  }
+
   .small {
```
Comment (lightme-fan) [inline src/components/InputName/InputName.tsx:54]: Please, use styled component.

File: src/components/InputName/InputName.tsx [hunk 7]
```diff
@@ -42,17 +60,6 @@ const InputContainer = styled.div`
...
-    line-height: 19px;
-    display: flex;
-    color: #979797;
-    padding-bottom: 6px;
-    align-items: start;
-    font-family: 'Futura Std', Arial, Helvetica, sans-serif;
-  }
 `
 
 const InputElement = styled.input`
```
Comment (lightme-fan) [inline src/components/InputName/InputName.tsx:65]: If there is nothing in this style, remove it.

**Review by lightme-fan — COMMENTED** [review]
Looking good!

**Review by lightme-fan — APPROVED** [review]

**Review by mitia-anah — APPROVED** [review]

File: src/components/Buttons/Buttons.tsx [hunk 8]
```diff
@@ -2,77 +2,64 @@ import React from 'react'
...
-          disabled={disabled}
-          onClick={() => ''}
-          {...props}>
-          {label}
-        </Button>
-      </SecondWrapper>
-    </FirstWrapper>
+    <ButtonStyle
+      disabled={false}
+      isPrimary={isPrimary ? true : false}
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:57]: you can just pass isPrimary

File: src/components/Buttons/Buttons.tsx [hunk 9]
```diff
@@ -2,77 +2,64 @@ import React from 'react'
...
-          learnEnabled={learnEnabled ? true : false}
-          disabled={disabled}
-          onClick={() => ''}
-          {...props}>
-          {label}
-        </Button>
-      </SecondWrapper>
-    </FirstWrapper>
+    <ButtonStyle
+      disabled={false}
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:56]: why false is hardcoded ?

File: src/components/RegisterForm/formRegister.tsx [hunk 10]
```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react'
```
Comment (andrzej-onja) [inline src/components/RegisterForm/formRegister.tsx:1]: align with naming convention

Comment (andrzej-onja) [reply src/components/RegisterForm/formRegister.tsx:1]: RegisterForm/formRegister.tsx 
RegisterForm/RegisterForm.tsx

File: src/components/RegisterForm/formRegister.tsx [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+  }
+  formType: {
+    onSubmit?: any
+  }
+}
+
+export const FormRegister: React.FC<FormProps> = ({
+  inputType,
+  button,
+  condition,
```
Comment (andrzej-onja) [inline src/components/RegisterForm/formRegister.tsx:40]: please implement props as discussed
onSubmit,
errors:

where errrors is an object of type  {password:String, email:string, username:string}

File: src/components/Buttons/Buttons.tsx [hunk 12]
```diff
@@ -24,39 +18,50 @@ const Button = styled.button<ButtonsProps>`
...
-        </Button>
-      </SecondWrapper>
-    </FirstWrapper>
+    <>
+      <ButtonStyled
+        disabled={disabled}
+        isPrimary={isPrimary}
+        onClick={() => {}}
+        {...props}>
+        {!isPrimary && <img src={icon} alt='icon' />}
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:61]: icons can also be in primary variant

File: src/components/RegisterForm/RegisterForm.tsx [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+  const [checked, setChecked] = React.useState<boolean>(false)
+  const [password, setPassword] = React.useState({
+    text: 'show',
+    inputPassword: '',
+    showPassword: false,
+  })
+
+  const [valid, setValid] = React.useState({
+    inputValue: '',
+    isValid: false,
```
Comment (andrzej-onja) [inline src/components/RegisterForm/RegisterForm.tsx:30]: isvalid unused, remove

File: src/components/RegisterForm/RegisterForm.tsx [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+  onSubmit,
+}) => {
+  const [checked, setChecked] = React.useState<boolean>(false)
+  const [password, setPassword] = React.useState({
+    text: 'show',
+    inputPassword: '',
+    showPassword: false,
+  })
+
+  const [valid, setValid] = React.useState({
```
Comment (andrzej-onja) [inline src/components/RegisterForm/RegisterForm.tsx:28]: I think this should be renamed from valid to something more general like inputInfo

File: src/components/RegisterForm/RegisterForm.tsx [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+        label='Your full name'
+        value={valid.userName}
+        onChange={() => {}}
+      />
+      <Input
+        type='text'
+        placeholder='Enter email adress'
+        label={'Email Adress'}
+        errorMes={errorMes}
+        value={valid.userName}
```
Comment (andrzej-onja) [inline src/components/RegisterForm/RegisterForm.tsx:86]: valid.userName ?? this is email

File: src/components/RegisterForm/RegisterForm.tsx [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+      ...password,
+      inputPassword: event.target.value,
+      text: 'hide',
+    })
+  }
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputInfo({
+      ...inputInfo,
+      correctEmail: true,
```
Comment (andrzej-onja) [inline src/components/RegisterForm/RegisterForm.tsx:54]: correctEmail is unused, remove

**Review by andrzej-onja — CHANGES_REQUESTED** [review]

File: src/components/Buttons/Buttons.tsx [hunk 17]
```diff
@@ -24,39 +18,50 @@ const Button = styled.button<ButtonsProps>`
...
+  box-shadow: -3px 3px rgba(21, 140, 177, 0.3), 3px -3px rgba(252, 70, 43, 0.3); 
 `
-const Buttons: React.FC<ButtonsProps> = ({
+const Button: React.FC<ButtonProps> = ({
   label,
   onClick,
   disabled,
-  learnEnabled,
+  isPrimary,
+  type,
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:50]: type is unused ?

File: src/components/Buttons/Buttons.tsx [hunk 18]
```diff
@@ -24,39 +18,50 @@ const Button = styled.button<ButtonsProps>`
...
+  icon,
   ...props
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:52]: remove ...props , explicitly pass all props

File: src/components/Buttons/Buttons.tsx [hunk 19]
```diff
@@ -1,21 +1,15 @@
 import React from 'react'
 import styled from 'styled-components'
 import { fonts } from '../../globalStyles/fonts'
-export interface ButtonsProps {
+export interface ButtonProps {
   label?: string
```
Comment (andrzej-onja) [inline src/components/Buttons/Buttons.tsx:5]: make label mandatory

File: src/components/RegisterForm/RegisterForm.tsx [hunk 20]
```diff
@@ -0,0 +1,172 @@
...
+            isPrimary={true}
+            onClick={() => {}}
+          />
+        </div>
+
+        <Buttons
+          type='button'
+          label='Register with Google'
+          isPrimary={false}
+          icon={icon}
```
Comment (andrzej-onja) [inline src/components/RegisterForm/RegisterForm.tsx:119]: hardcode icon we dont need to pass it as a prop to the form

**Review by andrzej-onja — APPROVED** [review]

**Pull request merged**